src/python/gudhi/wasserstein/wasserstein.py

```diff
@@ -125,20 +234,49 @@ def wasserstein_distance(X, Y, matching=False, order=1., internal_p=np.inf, enab
             if not matching:
                 return _perstot(Y, order, internal_p, enable_autodiff)
             else:
-                return _perstot(Y, order, internal_p, enable_autodiff), np.array([[-1, j] for j in range(m)])
+                cost = _perstot(Y, order, internal_p, enable_autodiff)
+                if cost == np.inf:  # We had some essential part here.
+                    return cost, None
+                else:
+                    return cost, np.array([[-1, j] for j in range(m)])
     elif m == 0:
         if not matching:
             return _perstot(X, order, internal_p, enable_autodiff)
         else:
-            return _perstot(X, order, internal_p, enable_autodiff), np.array([[i, -1] for i in range(n)])
+            cost = _perstot(X, order, internal_p, enable_autodiff)
+            if cost == np.inf:
+                return cost, None
+            else:
+                return cost, np.array([[i, -1] for i in range(n)])
 
+
+    # Second step: handle essential parts
+    if keep_essential_parts:
```

Comment:
The PR mentions a warning when keep_essential_parts&enable_autodiff, where is it?

Reply:
Done